src/syrupy/assertion.py

```diff
@@ -39,14 +51,25 @@ def __init__(
     def serializer(self) -> "AbstractSnapshotSerializer":
         if not getattr(self, "_serializer", None):
             self._serializer: "AbstractSnapshotSerializer" = self._serializer_class(
-                test_location=self._test_location, file_hook=self._file_hook
+                test_location=self._test_location
             )
         return self._serializer
 
     @property
     def num_executions(self) -> int:
         return int(self._executions)
 
+    @property
+    def executions(self) -> Dict[int, AssertionResult]:
+        return self._execution_results
+
+    @property
+    def discovered_snapshots(self) -> "SnapshotFiles":
+        return {
```

Comment:
What's the reasoning for moving this into the assertion class?

Reply:
It's pretty much the same logic, but now each assertion has a reference to the snapshots it discovers